File: Jessica/38.Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n==1:
            return '1'
        if n==2:
            return '11'
        currentStr='11'        
        for i in range (3,n+1):            
            count=1
            ret=''
            for j in range(1,len(currentStr)):
                if currentStr[j]==currentStr[j-1]:
                    count+=1
                else:                    
                    ret+=str(count)+currentStr[j-1]
                    count=1                    
                if j==len(currentStr)-1:
                    ret+=str(count)+currentStr[j]
            currentStr=ret
        return ret
                    
            
            
            
            
        

File: Jessica/38.Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
       if(n==1)
            return "1";
        if(n==2)
            return "11";
        string currentStr="11";
        string ret=""; 
        for(int i=3;i<=n;i++){
           
            int count=1;
            ret="";    
            for(int j=1;j<currentStr.length();j++){
           
                if(currentStr[j] != currentStr[j-1]){
                    ret+=to_string(count)+currentStr[j-1];
                    count=1;
                }
                   
                else
                    count++;
                if(j==currentStr.length()-1)
                        ret+=to_string(count)+currentStr[j];
                else
                     continue;
                    
                
            }
            
           currentStr=ret;
            
        }
        
        return ret;
        
    }
};
